Adamant/Modules/Chat/View/Subviews/ChatBaseMessage/ChatMessageCell.swift

```diff
@@ -491,47 +492,17 @@ private extension ChatMessageCell {
     @objc func tapReactionAction() {
         chatMenuManager.presentMenuProgrammatically(for: containerView)
     }
-    
-    @objc private func handleLongPressToCopy(_ gesture: UILongPressGestureRecognizer) {
-        switch gesture.state {
-        case .began:
-            didCopy = false
-            messageContainerView.animatePressDown()
-            
-            Task { [weak self] in
-                try? await Task.sleep(interval: quickCopyInterval)
-                
-                guard let self = self,
-                      gesture.state == .began || gesture.state == .changed else { return }
-                
-                await MainActor.run {
-                    self.longPressCopyAction()
-                    self.didCopy = true
-                }
-            }.stored(in: taskManager)
-            
-        case .ended:
-            if !didCopy {
-                taskManager.clean()
-                longPressCopyAction()
-            }
-            
-        case .cancelled, .failed:
-            messageContainerView.animatePressUp()
-            
-        default:
-            break
-        }
-    }
-    
-    private func longPressCopyAction() {
-        self.messageContainerView.animatePressUp()
-        UIPasteboard.general.string = self.model.text.string
-        self.copyNotification?()
-    }
 }
 
 extension ChatMessageCell: ChatMenuManagerDelegate {
+    var isFailedMessage: Bool {
+        model.backgroundColor == .failed
```

Comment:
Generally speaking ,deciding on the state based on backgroundColor is not a future-proof solution. Would you consider improving this part a bit?